[Solved #84]  create start-race controller

supunTE opened the pull request:
Solved #84 
- Create util for generating race text
- Client mock update (create a function to emit start race)
- Enhance type declaration in logger.
- Update start countdown protocol docs
- **Create start-race controller**
- Implement validations to check player's tournament using the socket room

supunTE committed:
refactor(socket protocols): refactor socket protocols to an object

supunTE committed:
fix(package.json): fix uuid package not imported into package.json by installing it

supunTE committed:
fix(store/effects): fix typo

change endCoundown to endCountdown

supunTE committed:
fix(store/effects): fix typo

change endCoundown to endCountdown

supunTE committed:
feat(utils): create a server util for generate race text

supunTE committed:
feat(utils): create a server util for generate race text

supunTE committed:
build(package.json): install axios

supunTE committed:
build(package.json): install axios

supunTE committed:
feat(client/room): add button to send startRace message

supunTE committed:
feat(client/room): add button to send startRace message

supunTE committed:
refactor(server/logger): enhance types for additional data for logger

supunTE committed:
refactor(server/logger): enhance types for additional data for logger

supunTE committed:
feat(client): add start race emitter to client

supunTE committed:
feat(client): add start race emitter to client

supunTE committed:
docs(protocols): update start countdown protocol

supunTE committed:
docs(protocols): update start countdown protocol

supunTE committed:
feat(server/controllers): create start race controller

supunTE committed:
Merge branch 'feature/82-implement-socket-schema-validation' into feature/84-start-race-controller

supunTE committed:
Merge branch 'feature/82-implement-socket-schema-validation' into feature/84-start-race-controller

dalanad commented on apps/server/src/utils/generateRaceText.ts:
```diff
@@ -0,0 +1,14 @@
+import axios from 'axios';
+
+export const generateRaceText = async (): Promise<string> => {
```
shouldn't this be parameterized to include paragraph count and other attributes? of text

supunTE replied:
Currently, we only need 1 paragraph for race. we can expand it later.

dalanad commented on apps/server/src/controllers/start-race.controller.ts:
```diff
@@ -0,0 +1,72 @@
+  context,
+  playerId,
+  tournamentId,
+}: StartRaceRequestArgs): Promise<void> => {
```
As we have already discussed shall we send  error messages back to the user

supunTE replied:
Opened a separate issue for this https://github.com/the-ai-team/razor/issues/80

supunTE committed:
Merge branch 'feature/82-implement-socket-schema-validation' into HEAD

supunTE committed:
feat(server): implement socket schema validation

 Conflicts:
	apps/server/src/models/protcol-schema-map.ts
	apps/server/src/services/main-utils/publish-on-receive.ts
	libs/models/src/lib/sockets/protocol-schemas.ts

supunTE committed:
feat(server/controllers): create start race controller

 Conflicts:
	apps/server/src/models/protcol-schema-map.ts
	apps/server/src/services/main-utils/publish-on-receive.ts
	libs/models/src/lib/sockets/protocol-schemas.ts

supunTE committed:
Merge remote-tracking branch 'origin/feature/84-start-race-controller' into feature/84-start-race-controller

supunTE committed:
fix: fix typo
